### ISSUE TITLE
[turborepo] - git diff misses changes when not ancestor

### ISSUE THREAD

**Issue by jgoldsmith613** [vercel/turborepo#3432]
### What version of Turborepo are you using?

1.5.5

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

When you do a filter of the form

pnpm turbo --filter="@package_name...[git_rev]"
If the git rev is not an ancestor of HEAD changes are missed.

It seems like in [git.go]( https://github.com/vercel/turbo/blob/main/cli/internal/scm/git.go#L24)
the git_rev becomes the from commit, and HEAD is the to commit.
So you are doing a diff on HEAD then a merge-base diff on the commit i'm passing in, which misses changes if they are part of a commit after the branches diverged.

Although not quite the same its similar to:[existing Issue](https://github.com/vercel/turbo/issues/514)

### Expected Behavior

Have a git diff based directly off the commit im passing in.

Basically like the issue linked above describes.

### To Reproduce

create a branch off main and make some changes.

create a separate branch off main and make different changes.

Do a git diff from the commit sha of HEAD from the initial branch.  Those changes will be missed.

**Comment by chris-olszewski** [comment]
Checked in latest canary and this has been resolved!
